### PR TITLE
Add NFS server to Asset Manager

### PR DESCRIPTION
**Pull request by kentsanggds**
## What

Adds NFS server to Asset Manager to enable upload and virus scanning of files before moving passing files to s3

## Reference

https://trello.com/c/pRBFWfGx/688-asset-manager-replatforming

### CODE REVIEW

**Review by bilbof — APPROVED** [review]
LGTM

File: charts/asset-manager/templates/deployment.yaml [hunk 1]
```diff
@@ -94,3 +97,8 @@ spec:
       - name: {{ .Release.Name }}-nginx-conf
         configMap:
           name: {{ .Release.Name }}-nginx-conf
+      - name: {{ .Release.Name }}-asset-uploads-efs
+        nfs: 
+          server: "{{ required "An NFS server is required!" .Values.nfsServer }}"
+          path: /
+
```
Comment (sengi) [inline charts/asset-manager/templates/deployment.yaml:104]: tiny nit: something funny going on with the line endings here (looks like a blank line followed by a missing line ending at EOF, somehow - check your editor config?)

File: charts/asset-manager/values.yaml [hunk 2]
```diff
@@ -71,6 +71,9 @@ clamResources:
     cpu: 500m
     memory: 1Gi
 
+# get NFS server from AWS console, overridden in ArgoCD config.
```
Comment (sengi) [inline charts/asset-manager/values.yaml:74]: Helm value comments follow the same convention as Go comments in that they start with the name of the variable (mostly for the benefit of automatic doc-generating tools).
```suggestion
# nfsServer is the address of the NFSv4 (or Amazon EFS) server where uploaded
# assets files are stored temporarily while awaiting virus scanning. Required.
# Overridden in ArgoCD config.
```

**Review by sengi — APPROVED** [review]
Neat. LGTM - just a couple of small suggestions to improve it further 👍

File: charts/argocd-apps/values-integration.yaml [hunk 3]
```diff
@@ -684,6 +684,7 @@ applications:
       - name: asset-manager.eks.integration.govuk.digital
     replicaCount: 1
     workerReplicaCount: 1
+    nfsServer: fs-b0dd6679.efs.eu-west-1.amazonaws.com
```
[the remark this answers is not in the document]
Comment (kentsanggds) [reply charts/argocd-apps/values-integration.yaml:687]: 👍

File: charts/asset-manager/values.yaml (hunk 2 above)
Comment (kentsanggds) [reply charts/asset-manager/values.yaml:74]: 👍

File: charts/asset-manager/templates/deployment.yaml (hunk 1 above)
Comment (kentsanggds) [reply charts/asset-manager/templates/deployment.yaml]: 👍

**Pull request merged**